homework-submission/barbarosapp/src/App.js

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from "react";
+import "./App.css";
+import logo from "./logo.svg";
+import ItemList from "./components/itemList";
+import Data from "./data/data";
+import AddMenu from "./components/addMenu";
+
+class App extends Component {
+  state = { Data };
+
+  handleCheck = id => {
+    const newData = this.state.Data.map(item => {
+      if (item.id === id) {
+        return { ...item, done: !item.done };
+      }
+      return item;
+    });
+
+    this.setState({ Data: newData });
+  };
+
+  handleRemove = id => {
+    function filterByID(item) {
+      if (item.id !== id) {
+        return true;
+      }
+      return false; 
+    }
+
+    const deletedData = this.state.Data.filter(filterByID);
+    this.setState({ Data: deletedData });
+  };
+
+  handleAdd = (name, deadline) => {
+    const newId = this.state.Data.length;
```

Comment:
This is a `bug`

![ezgif com-video-to-gif 1](https://user-images.githubusercontent.com/25272822/48372710-b8e5a580-e6bf-11e8-9d97-f47650a234b9.gif)

Ok let's just break it down!

let's assume that we have three elements:
 1. two are already there with `id`s 1, 2
 2. I removed the first one, which have `1` as an `id` and added a new one

So with that we will have the following `id`s [2, 3], next we added a new two elements:
> according to the new length we will have the following `id`s

[2, 3, 1, 2] if I added a new one I will fil in the blank the second `3` which is already there, don't you agree?

------
for solving this it's probably be better to stay away from the `.length` property, and maybe using another methodology that may solve too many things, like `uuid` or something other. 🙂

Reply:
Yes, I can create items with same id's. So i will try to use guid